### ISSUE TITLE
JS error when ordering an entity table with infinite scroll pagination

### ISSUE THREAD

**Issue by jdubois** [jhipster/generator-jhipster#2422]
This is on the current master branch (unreleased v2.25.0), with the new ordering of entities ( @deepu105 I think it comes from you? ).

- Generate a new entity
- Select "infinite scroll" pagination
- Do a lot of entities
- Order those entity on an attribute (not the ID)

You will get the following error:

    Error: [ngRepeat:dupes] Duplicates in a repeater are not allowed. Use 'track by' expression to specify
    unique keys. Repeater: buzz in buzzs track by buzz.id, Duplicate key: 61, Duplicate value:
    {"id":61,"aaa":"ttttt"}

**Comment by deepu105** [comment]
The error is strange, id should not be repeating and we already have track by id.
Anyway @geraldhumphries since the PR was from you can you plz take a look?

**Comment by geraldhumphries** [comment]
Looking into it now. I remember seeing this error but I thought I fixed it.

**Comment by geraldhumphries** [comment]
I was able to reproduce. I inserted a bunch of rows with the same values in one of the fields, then added a few more rows with different values. When I sorted by that field and started scrolling down, I got duplicate rows at the end leading to the Angular error. Running the query it uses in Postgres directly caused the same thing to happen. I found a SO question that answers why:

stackoverflow.com/questions/13580826/postgresql-repeating-rows-from-limit-offset

I think the solution is to always sort by id as a secondary sorting column.
